curtsies/events.py

```diff
@@ -191,76 +220,104 @@ def get_key(bytes_, encoding, keynames='curtsies', full=False):
             if seq in CURTSIES_NAMES:
                 return CURTSIES_NAMES[seq]
-            return seq.decode(encoding) #assumes that curtsies names are a subset of curses ones
+            return seq.decode(
+                encoding
+            )  # assumes that curtsies names are a subset of curses ones
         else:
-            assert keynames == 'bytes'
+            assert keynames == "bytes"
             return seq  # type: ignore
 
-    key_known = seq in CURTSIES_NAMES or seq in CURSES_NAMES or decodable(seq, encoding)
+    key_known = (
+        seq in CURTSIES_NAMES or seq in CURSES_NAMES or decodable(seq, encoding)
+    )
 
     if full and key_known:
         return key_name()
     elif seq in KEYMAP_PREFIXES or could_be_unfinished_char(seq, encoding):
-        return None # need more input to make up a full keypress
+        return None  # need more input to make up a full keypress
     elif key_known:
         return key_name()
     else:
-        seq.decode(encoding) # this will raise a unicode error (they're annoying to raise ourselves)
-        assert False, 'should have raised an unicode decode error'
+        seq.decode(
+            encoding
+        )  # this will raise a unicode error (they're annoying to raise ourselves)
+        assert False, "should have raised an unicode decode error"
+
 
 def could_be_unfinished_char(seq, encoding):
     # type: (bytes, Text) -> bool
     """Whether seq bytes might create a char in encoding if more bytes were added"""
     if decodable(seq, encoding):
-        return False # any sensible encoding surely doesn't require lookahead (right?)
+        return False  # any sensible encoding surely doesn't require lookahead (right?)
         # (if seq bytes encoding a character, adding another byte shouldn't also encode something)
 
-    if codecs.getdecoder('utf8') is codecs.getdecoder(encoding):
+    if codecs.getdecoder("utf8") is codecs.getdecoder(encoding):
         return could_be_unfinished_utf8(seq)
-    elif codecs.getdecoder('ascii') is codecs.getdecoder(encoding):
+    elif codecs.getdecoder("ascii") is codecs.getdecoder(encoding):
         return False
     else:
-        return True # We don't know, it could be
+        return True  # We don't know, it could be
+
 
 def could_be_unfinished_utf8(seq):
     # type: (bytes) -> bool
     # http://en.wikipedia.org/wiki/UTF-8#Description
-    if   ord(seq[0:1]) & 0b10000000 == 0b10000000 and len(seq) < 1: return True
-    elif ord(seq[0:1]) & 0b11100000 == 0b11000000 and len(seq) < 2: return True
-    elif ord(seq[0:1]) & 0b11110000 == 0b11100000 and len(seq) < 3: return True
-    elif ord(seq[0:1]) & 0b11111000 == 0b11110000 and len(seq) < 4: return True
-    elif ord(seq[0:1]) & 0b11111100 == 0b11111000 and len(seq) < 5: return True
-    elif ord(seq[0:1]) & 0b11111110 == 0b11111100 and len(seq) < 6: return True
-    else: return False
+    if ord(seq[0:1]) & 0b10000000 == 0b10000000 and len(seq) < 1:
```

Comment:
I think having these lines close together so you can see the bit patterns is important, let's ignore these 10 lines or so